src/classes/state/state-cache-storage.ts

```diff
@@ -25,16 +25,32 @@ const unlinkSafely = (filePath: string) => {
   }
 };
 
-const resetCacheWithOctokit = async (cacheKey: string): Promise<void> => {
+const getOctokitClient = () => {
   const token = core.getInput('repo-token');
-  const client = getOctokit(token, undefined, octokitRetry);
-  // TODO: better way to get repository?
-  const repo = process.env['GITHUB_REPOSITORY'];
+  return getOctokit(token, undefined, octokitRetry);
+};
+
+const checkIfCacheExists = async (cacheKey: string): Promise<boolean> => {
+  const client = getOctokitClient();
+  try {
+    const issueResult = await client.request(
+      `/repos/${context.repo.owner}/${context.repo.repo}/actions/caches`
+    );
+    const caches: Array<{key?: string}> =
+      issueResult.data['actions_caches'] || [];
+    return Boolean(caches.find(cache => cache['key'] === cacheKey));
+  } catch (error) {
+    core.debug(`$Error checking if cache exist: ${error.message}`);
```

Comment:
```suggestion
    core.debug(`Error checking if cache exist: ${error.message}`);
```
Excessive dollar sign?